api/handlers.go

```diff
@@ -211,60 +211,78 @@ func (s *Server) JetDropsByJetID(ctx echo.Context, jetID server.JetIdPath, param
+
+	cnt := int64(total)
+	drops := make([]server.JetDrop, len(jetDrops))
+	for i, jetDrop := range jetDrops {
+		nextJetDrops := []server.NextPrevJetDrop{}
+		prevJetDrops := []server.NextPrevJetDrop{}
+
+		prev, ok := jetDropsByHash[base64.StdEncoding.EncodeToString(jetDrop.FirstPrevHash)]
+		if ok {
+			prevJetDrops = append(prevJetDrops, prev)
 		}
-		_, prevJetDropsLastElem, nextJetDropsLastElem, err = s.storage.GetJetDropByID(
-			models.JetDropID{
-				JetID:       jetDrops[len(jetDrops)-1].JetID,
-				PulseNumber: jetDrops[len(jetDrops)-1].PulseNumber,
-			})
-		if err != nil {
-			return ctx.JSON(http.StatusInternalServerError, "imposible situation")
+		prev, ok = jetDropsByHash[base64.StdEncoding.EncodeToString(jetDrop.SecondPrevHash)]
+		if ok {
+			prevJetDrops = append(prevJetDrops, prev)
 		}
 
-		logCustomFields := func(logFormatFunction func(fmt string, args ...interface{}), jetDrops []models.JetDrop) {
-			custom := make([]models.JetDrop, len(jetDrops))
-			for i, jetDrop := range jetDrops {
-				custom[i] = models.JetDrop{
-					PulseNumber:  jetDrop.PulseNumber,
-					JetID:        jetDrop.JetID,
-					Timestamp:    jetDrop.Timestamp,
-					RecordAmount: jetDrop.RecordAmount,
-				}
-			}
-			logFormatFunction("%+v", custom)
+		next, ok := jetDropsByPrevHash[base64.StdEncoding.EncodeToString(jetDrop.Hash)]
+		if ok {
+			nextJetDrops = append(nextJetDrops, next...)
+		}
+
+		drops[i] = JetDropToAPI(jetDrop, prevJetDrops, nextJetDrops)
+	}
+	return ctx.JSON(http.StatusOK, server.JetDropsResponse{
+		Total:  &cnt,
+		Result: &drops,
+	})
+}
+
+func (s *Server) getEnrichingJetDrops(oldestPulse, newestPulse int64) ([]models.JetDrop, error) {
+	var enrichedDrops []models.JetDrop
+	emptyPulse := models.Pulse{}
+
+	pulse, err := s.storage.GetPulse(oldestPulse)
+	if err == nil && pulse.PrevPulseNumber > 0 {
+		ejd, err := s.storage.GetJetDrops(models.Pulse{PulseNumber: pulse.PrevPulseNumber})
+		if err != nil {
+			return []models.JetDrop{}, errors.Wrapf(err, "can't enrich jetDrops for pulse %d from storage, from gte", pulse.PrevPulseNumber)
 		}
-		logCustomFields(s.logger.Debugf, prevJetDropsFirstElem)
-		logCustomFields(s.logger.Debugf, prevJetDropsLastElem)
-		logCustomFields(s.logger.Debugf, nextJetDropsLastElem)
-		logCustomFields(s.logger.Debugf, nextJetDropsFirstElem)
+		enrichedDrops = append(enrichedDrops, ejd...)
 	}
 
-	enrichJetDrops := func(jd []models.JetDrop) []models.JetDrop {
-		t := append(append(jd, prevJetDropsFirstElem...), nextJetDropsLastElem...)
-		return append(append(t, prevJetDropsLastElem...), nextJetDropsFirstElem...)
+	nextPulse, err := s.storage.GetNextSavedPulse(models.Pulse{
+		PulseNumber: newestPulse,
+	})
+	if err == nil && nextPulse != emptyPulse {
+		ejd, err := s.storage.GetJetDrops(models.Pulse{PulseNumber: nextPulse.PulseNumber})
+		if err != nil {
+			return []models.JetDrop{}, errors.Wrapf(err, "can't enrich jetDrops for pulse %d from storage, from lte", nextPulse.PulseNumber)
```

Comment:
`from lte ` seems strange here

Reply:
according to abobe, I can suggest "can't enrich jetDrops for next pulse %d"